jobs/build/ocp4/Jenkinsfile

```diff
@@ -119,9 +121,30 @@ node {
                 stage("initialize") { build.initialize() }
                 buildlib.assertBuildPermitted(doozerOpts)
                 stage("build RPMs") { build.stageBuildRpms() }
-                lock("compose-lock-${params.BUILD_VERSION}") {
-                    stage("build compose") { build.stageBuildCompose() }
+
+                // if the automation is not frozen perform compose
+                // otherwise if automation is frozen but there
+                // are rpms in the build plan perform compose
+                // and announce on slack
+
+                if(buildlib.isBuildPermitted(doozerOpts)){
+                    lock("compose-lock-${params.BUILD_VERSION}") {
+                        stage("build compose") { build.stageBuildCompose() }
+                    }
+                } else if(params.BUILD_RPMS != "none"){
```

Comment:
ocp4 by default checks whether artifacts need building. so even if RPMs are specified in `BUILD_RPMS`, they might not have been built. there's a build plan recorded in the `build` lib that we can read to find out if any RPMs actually got built.

isn't this a fun job?

Reply:
I am new to groovy. Infact need to take sometime out to learn it. I know I can use functions from imported groovy modules (like buildlib, commonlib). Can I you buildPlan the same way with .(dot) notation?

Reply:
try it and see; i don't actually know.

if not, you would need to pass back what you need from a subroutine call. i think i'd probably prefer that anyway, just for decoupling.